### DISCUSSION TITLE
API_KEY_SERVICE_BLOCKED

### DISCUSSION THREAD

**Discussion by TiVoShane** [firebase/firebase-ios-sdk#14243] [Q&A] [answered]
As of yesterday, my iOS app that's published to the App Store began getting this error. Does anyone have any insight as to what's happening and how I can fix it? I didn't make any changes on my end and I can't seem to figure out what API Key I would need to adjust, if any. It's unfortunately made my app unusable until this is fixed. 

11.6.0 - [FirebaseVertexAI][I-VTX002002] Response payload: {
  "error": {
    "code": 403,
    "message": "Requests to this API firebasevertexai.googleapis.com method google.firebase.vertexai.v1beta.PredictionService.GenerateContent are blocked.",
    "status": "PERMISSION_DENIED",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "API_KEY_SERVICE_BLOCKED",
        "domain": "googleapis.com",
        "metadata": {
          "consumer": "projects/624826303047",
          "methodName": "google.firebase.vertexai.v1beta.PredictionService.GenerateContent",
          "service": "firebasevertexai.googleapis.com",
          "apiName": "firebasevertexai.googleapis.com"
        }
      },
      {
        "@type": "type.googleapis.com/google.rpc.LocalizedMessage",
        "locale": "en-US",
        "message": "Requests to this API firebasevertexai.googleapis.com method google.firebase.vertexai.v1beta.PredictionService.GenerateContent are blocked."
      }
    ]
  }
}

**Comment by TiVoShane** [comment, marked as answer]
After discussing this eith a Google employee, I discovered I needed to go into the Google Cloud console and add the Vertex AI in Firebase API to the API Key.  I'm not sure why this changed overnight or what caused it if nothing changed, but it's working now.

console.cloud.google.com - APIs & Services - Credentials

**Reply by andrewheard** [reply]
Hi @TiVoShane, sorry about this issue. Typically the `Vertex AI in Firebase` API is automatically added to your allowlist during onboarding in the Firebase Console but we found, when enforcing this restriction, that in certain edge cases the API was not added to the allowlist.

**Comment by andrewheard** [comment]
For anyone running into this issue, please ensure that `Vertex AI in Firebase` is in the allowlist for your Firebase app's API key. We've added an [FAQ entry](https://firebase.google.com/docs/vertex-ai/faq-and-troubleshooting#error-requests-to-service-blocked) with more details on how to do this.
